src/lsi/lsi.py

```diff
@@ -322,42 +332,36 @@ def _run_ssh(entries, username, idfile, no_prompt=False, command=None,
                             show.extend(commands[1:])
                         _print_entries = True
                     else:
-                        print yellow('Unknown command "%s".' % cmd)
+                        print(yellow('Unknown command "%s".' % cmd))
         return _connect_ssh(entries[choice], username, idfile, tunnel)
 
 
 def _get_path(cmd):
     """Queries bash to find the path to a commmand on the system."""
     if cmd in _PATHS:
         return _PATHS[cmd]
-    proc = subprocess.Popen('which ' + cmd, shell=True, stdout=subprocess.PIPE)
-    out, err = proc.communicate()
-    if proc.wait() != 0:
-        raise IOError('Lookup of path to command {0} failed{1}'
-                      .format(repr(cmd), '' if err is None else ': ' + err))
-    _PATHS[cmd] = out.strip()
+    out = subprocess.check_output('which {}'.format(cmd), shell=True)
+    _PATHS[cmd] = out.decode("utf-8").strip()
     return _PATHS[cmd]
 
 
 def _build_ssh_command(hostname, username, idfile, ssh_command, tunnel):
     """Uses hostname and other info to construct an SSH command."""
     command = [_get_path('ssh'),
                '-o', 'StrictHostKeyChecking=no',
-               '-o', 'ConnectTimeout=5',
-               '-o', 'UserKnownHostsFile={}'.format(_KNOWN_HOSTS_FILE)]
+               '-o', 'ConnectTimeout=5']
```

Comment:
As long as we're editing this file could you bump the timeout to 10 seconds?  I get banner timeouts semi-regularly on some of our slow/busy machines